auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,410 @@
+
+    sdf::CustomInertiaCalcProperties calcInterface = CustomInertiaCalcProperties(
+      _density, *this, _autoInertiaParams);
+
+    return customCalculator(_errors, calcInterface);
+  }
+```
+
+### Configuring the `CalculateInertial()` function behavior
+
+`CalculateInertial()` functions are also added to the `sdf::Root`, `sdf::World`, 
+`sdf::Model`, `sdf::Link`, `sdf::Collision` and `sdf::Geometry` classes. Calling the 
+`CalculateInertial()` function of any class, in turn recursively calls the `CalculateInertial()` 
+for each element below it down the chain. For eg, generally a user might call the `Root::CalculateInertial()` 
+which further calls `World::CalculateInertials()` for all the worlds which in turn 
+calls the `Model::CaluclateInertials()` for all the models in the world and so on until 
+the final `CalculateInertial()` for the respective geometry type is called where the chain ends.
+
+The `sdf::ParserConfig` object is sent down the `CalculateInertial()` chain as a function 
+parameter and is used to get the configuration information like the registered Custom 
+Inertia Calculator. An `enum class` would be added to the `sdf::ParserConfig` with values 
+that would allow the user to configure the behavior of the `CalculateInertial()` function:
+
+```C++
+/// \enum ConfigureCalculateInertial
+/// \brief Configuration options of how CalculateInertial() function
+/// would be used
+enum class ConfigureCalculateInertial
+{
+  /// \brief If this value is used, CalculateInertial() won't be 
+  /// called from inside the Root::Load() function
+  SKIP_CALCULATION_IN_LOAD,
+
+  /// \brief If this values is used, CalculateInertial() would be 
+  /// called and the computed inertial values would be saved 
+  SAVE_CALCULATION
+};
+```
+
+Setting values from the above enum for the `sdf::ParserConfig` object, the user can 
+configure the `CalculateInertial()` functions. For eg: if the configuration is set to 
+`SKIP_CALCULATION_IN_LOAD` (which would be the default configuration), then the 
+`Root::CalculateInertial()` won't be called from within the `Root::Load()`. The user 
+would need to call the `Root:CalculateInertial()` separately after the load is complete. 
+This is also recommended as the inertia calculation uses the `PoseGraph` to resolve the 
+inertial poses for different collisions if they are not in the link frame. 
+
+## Proposed Mesh Inertia Calculation Methods
+
+### Voxelization-based method
+
+Voxels are the 3D equivalent of a pixel in 2D<sup>[\[1\]](#References)</sup>. Voxels can 
+be arranged in ‘Voxel Grids’ <sup>[\[2, 3\]](#References)</sup> which are the 3D equivalent 
+of a structured image using pixels. 
+
+During the voxelization of a point cloud or mesh, all the points in the 3D data are intersected 
+with a Voxel Grid. The voxels which contain a point of the mesh are kept while others are zeroed 
+out(discarded). This way, we are left with a voxelized mesh that closely resembles the original mesh.
```

Comment:
```suggestion
out (discarded). This way, we are left with a voxelized mesh that closely resembles the original mesh.
```

Reply:
Corrected in [fca06ae](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/fca06ae64a6843311fa125ff89c6f9a376c37599)